src/Raven.Server/Documents/StreamsTempFile.cs

```diff
@@ -161,6 +164,24 @@ private void AssertGeneration()
                 if (_parent.Generation != _generation)
                     throw new NotSupportedException($"Invalid generation. Parent: {_parent.Generation}. Current: {_generation}");
             }
+
+            public IDisposable CreateReaderStream(out LimitedStream limitedStream)
+            {
+                var safeFileHandle = new SafeFileHandle(_parent._file.InnerStream.SafeFileHandle.DangerousGetHandle(), ownsHandle: false);
+                Stream stream = new FileStream(safeFileHandle, FileAccess.Read);
+                stream.Seek(_startPosition, SeekOrigin.Begin);
+                if (_stream is TempCryptoStream tcs)
+                {
+                    stream = new TempCryptoStream(stream, tcs);
+                }
+                limitedStream = new LimitedStream(stream, Length, _startPosition, _startPosition);
+
+                return new DisposableAction(() =>
+                {
+                    stream.Dispose();
+                   safeFileHandle.Dispose();
```

Comment:
formatting

Reply:
can disposal of stream fail? maybe we should wrap those in usings to be sure?

```
using (safeFileHandle)
using (stream)
{
    // disposing
}
```

Reply:
> formatting

Done



> can disposal of stream fail? maybe we should wrap those in usings to be sure?
> 
> ```
> using (safeFileHandle)
> using (stream)
> {
>     // disposing
> }
> ```

Maybe I'm missing something, but I don't see how this can fail... we are calling it when disposing the batch items & streams

Reply:
let's wrap it in usings, its safer to do so, mind the order of disposal please :)

Reply:
Done